crates/primitives/src/transaction/mod.rs

```diff
@@ -612,6 +614,93 @@ impl From<TxEip4844> for Transaction {
     }
 }
 
+impl TryFrom<reth_rpc_types::Transaction> for Transaction {
+    type Error = ConversionError;
+
+    fn try_from(tx: reth_rpc_types::Transaction) -> Result<Self, Self::Error> {
+        match tx.transaction_type {
+            None => {
+                // legacy
+                Ok(Transaction::Legacy(TxLegacy {
+                    chain_id: tx.chain_id,
+                    nonce: tx.nonce,
+                    gas_price: tx.gas_price.ok_or(ConversionError::MissingGasPrice)?,
+                    gas_limit: tx
+                        .gas
+                        .try_into()
+                        .map_err(|_| ConversionError::Eip2718Error(RlpError::Overflow.into()))?,
+                    to: tx.to.map_or(TransactionKind::Create, TransactionKind::Call),
+                    value: tx.value,
+                    input: tx.input,
+                }))
+            }
+            Some(1u8) => {
+                // eip2930
+                Ok(Transaction::Eip2930(TxEip2930 {
+                    chain_id: tx.chain_id.ok_or(ConversionError::MissingChainId)?,
+                    nonce: tx.nonce,
+                    gas_limit: tx
+                        .gas
+                        .try_into()
+                        .map_err(|_| ConversionError::Eip2718Error(RlpError::Overflow.into()))?,
+                    to: tx.to.map_or(TransactionKind::Create, TransactionKind::Call),
+                    value: tx.value,
+                    input: tx.input,
+                    access_list: tx.access_list.ok_or(ConversionError::MissingAccessList)?.into(),
+                    gas_price: tx.gas_price.ok_or(ConversionError::MissingGasPrice)?,
+                }))
+            }
+            Some(2u8) => {
+                // EIP-1559
+                Ok(Transaction::Eip1559(TxEip1559 {
+                    chain_id: tx.chain_id.ok_or(ConversionError::MissingChainId)?,
+                    nonce: tx.nonce,
+                    max_priority_fee_per_gas: tx
+                        .max_priority_fee_per_gas
+                        .ok_or(ConversionError::MissingMaxPriorityFeePerGas)?,
+                    max_fee_per_gas: tx
+                        .max_fee_per_gas
+                        .ok_or(ConversionError::MissingMaxFeePerGas)?,
+                    gas_limit: tx
+                        .gas
+                        .try_into()
+                        .map_err(|_| ConversionError::Eip2718Error(RlpError::Overflow.into()))?,
+                    to: tx.to.map_or(TransactionKind::Create, TransactionKind::Call),
+                    value: tx.value,
+                    access_list: tx.access_list.ok_or(ConversionError::MissingAccessList)?.into(),
+                    input: tx.input,
+                }))
+            }
+            Some(3u8) => {
+                // EIP-4844
+                Ok(Transaction::Eip4844(TxEip4844 {
```

Comment:
3u8 should be the type for Eip4844?  The document here should be updated?
https://github.com/alloy-rs/alloy/blob/main/crates/rpc-types/src/eth/transaction/mod.rs#L106C5-L116C38

Reply:
yes, please update those docs